src/services/services.ts

```diff
@@ -944,6 +944,10 @@ namespace ts {
         }
     }
 
+    export interface HostCancellationToken {

```

Comment:
We're calling it "HostCancellationToken" to distinguish it from CancellationToken.  The latter is what we use internally with all our features.  The former is what we get from the host, and what the latter uses to actually make its determinations.

They're not hte same object because the former is used purely for knowing if something should cancel.  The latter is actually responsible for doing the checking *and* throwing the appropriate script side exception (something the host knows nothing about).